packages/flutter/lib/src/material/tabs.dart

```diff
@@ -381,19 +381,61 @@ class _TabsScrollBehavior extends BoundedBehavior {
   }
 }
 
-class TabBarSelection {
-  TabBarSelection({ int index: 0, this.maxIndex, this.onChanged }) : _index = index {
+class _TabBarSelection extends InheritedWidget {
+  _TabBarSelection({
+    this.selection,
+    Key key,
+    Widget child
+  }) : super(key: key, child: child);
+
+  final TabBarSelectionState selection;
+
+  bool updateShouldNotify(_TabBarSelection oldWidget) => selection != oldWidget.selection;
+}
+
+class TabBarSelection extends StatefulComponent {
+  TabBarSelection({
+    Key key,
+    this.index: 0,
+    this.maxIndex,
```

Comment:
I agree that a length (count) would be more natural here. I'm going to be replacing TabBarSelection with TabBarSelection<t> in my next commit, per https://github.com/flutter/flutter/pull/990.  So, if it's OK with you, I'd rather just leave maxIndex alone for now.